src/hotspot/share/opto/split_if.cpp

```diff
@@ -427,6 +409,27 @@ bool PhaseIdealLoop::clone_cmp_down(Node* n, const Node* blk1, const Node* blk2)
   return false;
 }
 
+// 'n' could be a node belonging to a Template Assertion Predicate Expression (i.e. any node between a Template
+// Assertion Predicate and its OpaqueLoop* nodes (included)). We cannot simply split this node up since this would
+// create a phi node inside the Template Assertion Predicate Expression - making it unrecognizable as such. Therefore,
+// we completely clone the entire Template Assertion Predicate Expression "down". This ensures that we have an
+// untouched copy that is still recognized by the Template Assertion Predicate matching code.
+void PhaseIdealLoop::clone_template_assertion_predicate_expression_down(Node* node) {
+  if (!TemplateAssertionPredicateExpressionNode::is_valid(node)) {
```

Comment:
Covers previous `subgraph_has_opaque()` call to check whether this node is part of a Template Assertion Predicate Expression.